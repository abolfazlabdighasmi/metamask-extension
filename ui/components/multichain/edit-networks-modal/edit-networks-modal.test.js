import React from 'react';
import { renderWithProvider } from '../../../../test/jest';
import configureStore from '../../../store/store';
import mockState from '../../../../test/data/mock-state.json';
import { EditNetworksModal } from '.';

const mockSetShowTestNetworks = jest.fn();

jest.mock('../../../store/actions.ts', () => ({
  setShowTestNetworks: () => mockSetShowTestNetworks,
}));

const render = ({
  showTestNetworks = false,
  currentChainId = '0x5',
  providerConfigId = 'chain5',
  isUnlocked = true,
} = {}) => {
  const state = {
    metamask: {
      ...mockState.metamask,
      isUnlocked,
      providerConfig: {
        ...mockState.metamask.providerConfig,
        chainId: currentChainId,
        id: providerConfigId,
      },
      preferences: {
        showTestNetworks,
      },
      useRequestQueue: true,
    },
    activeTab: {
      origin,
    },
  };

  const store = configureStore(state);
  return renderWithProvider(<EditNetworksModal onClose={jest.fn()} />, store);
};
describe('EditNetworksModal', () => {
  it('renders properly', () => {
    const { container } = render();
    expect(container).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const { getByText } = render();
    expect(getByText('Edit networks')).toBeInTheDocument();
  });
});
