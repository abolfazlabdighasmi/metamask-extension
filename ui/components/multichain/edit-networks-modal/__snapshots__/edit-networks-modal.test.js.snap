// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNetworksModal renders properly 1`] = `<div />`;
